### DISCUSSION TITLE
Build with old JDKs

### DISCUSSION THREAD

**Discussion by AIGLEZMA** [wiverson/maven-jpackage-template#33] [Q&A] [answered]
Hey,
I'm using a library, but this library only works on java 8-9, is there any way I could get this template building native installers on those versions ?

Thank you,

**Comment by wiverson** [comment, marked as answer]
jpackage and jlink only became fully supported / non-preview in JDK 16. You could enable it on 15 but it's kind of a pain.

What / why does the library only work on 8/9?

There are other installer packages that work with JDKs that old, but AFAIK they will bundle a (relatively) large JVM in with the installer.

**Reply by AIGLEZMA** [reply]
I don't know why, but i'm not the only one who has that issue. (it's JFoenix btw)

**Reply by wiverson** [reply]
Huh. JFoenix looks like abandonware - the website shows the last update as 2017. Similarly the GitHub repo looks like it really hasn't been updated in years.  :P

JavaFX alternatives: https://www.pixelduke.com/open-source/
Not JavaFX (Swing) but you might find interesting: https://www.formdev.com/flatlaf/

**Reply by wiverson** [reply]
FWIW according to this page: https://www.formdev.com/flatlaf/themes/ you can use IntelliJ themes with FlatLaF, including a bunch of different Material design UIs.

**Reply by AIGLEZMA** [reply]
That is very interesting, thank you !
